clients/src/main/java/org/apache/kafka/common/security/authenticator/SaslClientAuthenticator.java

```diff
@@ -342,6 +342,7 @@ public void reauthenticate(ReauthenticationContext reauthenticationContext) thro
         reauthInfo.reauthenticating(apiVersionsResponseFromOriginalAuthentication,
                 reauthenticationContext.reauthenticationBeginNanos());
         NetworkReceive netInBufferFromChannel = reauthenticationContext.networkReceive();
+        resetNetworkIn(); // reset the buffer before new assignment
```

Comment:
Why do we need to do this?

Reply:
In the very next line, we are overwriting the `netInBuffer` member and hence losing it's earlier value without having closed it properly. Given the sensitivity of the change in network component here, I am removing all changes to this buffer from this PR to unblock it and will file a separate PR to address this.